src/EnergyPlus/UnitarySystem.cc

```diff
@@ -11053,6 +11053,9 @@ namespace UnitarySystems {
         if (this->m_IsZoneEquipment) {
             turnFansOn = state.dataHVACGlobal->ZoneCompTurnFansOn;
             turnFansOff = state.dataHVACGlobal->ZoneCompTurnFansOff;
+            // set global flag to act on the fan operation
+            state.dataHVACGlobal->TurnFansOn = turnFansOn;
+            state.dataHVACGlobal->TurnFansOff = turnFansOff;
```

Comment:
Wouldn't it be better to continue passing explicit flags to `SimulateFanComponents` instead of relying on a global signal?

Reply:
But I guess we're not doing that in very many places. My concern is whether the global flags are getting reset properly after a particular piece of zone equipment (like here) touches them.

Reply:
I checked and the flags are managed properly for zone and air loop equipment. Whether we should set a flag or pass a flag is another question. I think doing it this way is cleaner, I don't have to check whether the fan is in zone equipment before calling the fan object. This is the same way Fan:SystemModel is handeled (i.e., don't check for zone equipment) which created this issue. Of course, this is up for debate.

Reply:
Quick search shows that these flag variables are set in [SimAirServingZones](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/SimAirServingZones.cc#L2494-L2507) and [ZoneEquipmentManager](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/ZoneEquipmentManager.cc#L3105-L3122), elsewhere they are just.

Reply:
Except that we are resetting them in `UnitarySystem` now.

Reply:
Yeah, I think if these are set at line 3123 in ZoneEquipmentManger:

    state.dataHVACGlobal->TurnFansOn = TurnFansOn;
    state.dataHVACGlobal->TurnFansOff = TurnFansOff;

and also reset these before the check of SAMs at line 3104 the result would be the same. The parent object could then use these flags to control the fan, that is, for those zone objects that use this flag to control the fan.

Maybe this type of change should be tested in a different branch? I guess it's a pretty small change so maybe it's worth a try.

Reply:
As I think about this more, I don't think the availability manager flags need to be seperate for zone and air loop equipment since zone equipment is called in their own loop, and airloop equipment is called in their own loop. Inside each of these loops the availability manager status is checked. So using a single set of flags should be adequate.

I don't expect this to be addressed in this branch.

Zone equipment:

    state.dataHVACGlobal->ZoneCompTurnFansOn
    state.dataHVACGlobal->ZoneCompTurnFansOn

Air loop equipment:

    state.dataHVACGlobal->TurnFansOn
    state.dataHVACGlobal->TurnFansOff

Reply:
Where does `CycleOnZoneFansOnly` fit into this?

Reply:
IIRC, that is only used with zone exhaust fans.

Reply:
> IIRC, that is only used with zone exhaust fans.

And PIU fans.

Reply:
Yeah, maybe it's more complicated than first thought.

    bool PIUTurnFansOn =
        (state.dataHVACGlobal->TurnFansOn || state.dataHVACGlobal->TurnZoneFansOnlyOn); // If True, overrides fan schedule and cycles PIU fan on
    bool const PIUTurnFansOff = state.dataHVACGlobal->TurnFansOff; // If True, overrides fan schedule and PIUTurnFansOn and cycles PIU fan off

Reply:
Maybe it would be safer to save the correct avail manager flag in the UnitarySystem struct and always pass these 2 variables to the fan model? 

    if (this->isZoneEquipment) {
        this->availManagerTurnOn = state.dataHVACGlobal->ZoneCompTurnFansOn;
        this->availManagerTurnOff = state.dataHVACGlobal->ZoneCompTurnFansOff;
    } else {
        this->availManagerTurnOn = state.dataHVACGlobal->TurnFansOn;
        this->availManagerTurnOff = state.dataHVACGlobal->TurnFansOff;
    }

The PIU issue gives me some concern. What if there are 2 air loops, how is this single flag managed on the PIU call? It looks to me as if all zone equipment are simulated, and not by air loop. So I am wondering now if there is a problem with the PIU unit.

    for (int ControlledZoneNum = 1; ControlledZoneNum <= state.dataGlobal->NumOfZones; ++ControlledZoneNum) {

        auto &TurnFansOn = state.dataHVACGlobal->TurnFansOn;
        auto &TurnFansOff = state.dataHVACGlobal->TurnFansOff;
        auto &TurnZoneFansOnlyOn = state.dataHVACGlobal->TurnZoneFansOnlyOn;

            case ZoneEquip::AirDistUnit: { // 'ZoneHVAC:AirDistributionUnit'
                // Air loop system availability manager status only applies to PIU and exhaust fans
                // Check to see if System Availability Managers are asking for fans to cycle on or shut off
                // and set fan on/off flags accordingly.
                if (state.dataZoneEquip->ZoneEquipAvail(ControlledZoneNum) == DataHVACGlobals::CycleOn ||
                    state.dataZoneEquip->ZoneEquipAvail(ControlledZoneNum) == DataHVACGlobals::CycleOnZoneFansOnly) {
                    TurnFansOn = true;
                }
                if (state.dataZoneEquip->ZoneEquipAvail(ControlledZoneNum) == DataHVACGlobals::CycleOnZoneFansOnly) {
                    // Currently used only by parallel powered induction unit
                    TurnZoneFansOnlyOn = true;
                }
                if (state.dataZoneEquip->ZoneEquipAvail(ControlledZoneNum) == DataHVACGlobals::ForceOff) {
                    TurnFansOff = true;
                }

                ZoneAirLoopEquipmentManager::ManageZoneAirLoopEquipment(state,

Reply:
I see, each air loop sets ZoneEquipAvail for each zone attached to the air loop. And this value is used to set TurnFansOn/Off in the above code for each PIU. This still leads me to believe only 1 set of global flags are needed but not sure at this point.

SystemAvailabilityManager.cc

            // loop over the zones served by the system and set the zone equipment availability
            for (ZoneInSysNum = 1; ZoneInSysNum <= state.dataAirLoop->AirToZoneNodeInfo(PriAirSysNum).NumZonesCooled; ++ZoneInSysNum) {

                CtrldZoneNum = state.dataAirLoop->AirToZoneNodeInfo(PriAirSysNum).CoolCtrlZoneNums(ZoneInSysNum);
                state.dataZoneEquip->ZoneEquipAvail(CtrldZoneNum) = state.dataAirLoop->PriAirSysAvailMgr(PriAirSysNum).AvailStatus;
            }

Reply:
After what I learned on #9976, this is one correct fix. Using TurnFansOn/Off here will not impact any other equipment. This methodology is taken one step further in 9976 by removing these flags from fan calls. For now this works just fine.